services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/FlowOperationsBolt.java

```diff
@@ -16,30 +16,125 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class FlowOperationsBolt extends NeoOperationsBolt {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(FlowOperationsBolt.class);
+    private static final Logger logger = LoggerFactory.getLogger(FlowOperationsBolt.class);
 
     public FlowOperationsBolt(Auth neoAuth) {
         super(neoAuth);
     }
 
     @Override
     List<? extends InfoData> processRequest(Tuple tuple, BaseRequest request, Session session) {
-        LOGGER.warn("Received unsupported Flow Operation");
-        return null;
+        List<? extends InfoData> result = null;
+        if (request instanceof GetFlowIdsForLinkRequest) {
+            result = getFlowIdsForLink((GetFlowIdsForLinkRequest) request, session);
+        } else {
+            unhandledInput(tuple);
+        }
+
+        return result;
+    }
+
+    private List<FlowsResponse> getFlowIdsForLink(GetFlowIdsForLinkRequest request, Session session) {
+        SwitchId sourceSwitchId = request.getSource().getDatapath();
+        Integer sourcePort = request.getSource().getPortNumber();
+        SwitchId destinationSwitchId = request.getDestination().getDatapath();
+        Integer destinationPort = request.getDestination().getPortNumber();
+
+        if (sourceSwitchId == null || sourcePort == null || destinationSwitchId == null || destinationPort == null) {
+            logger.error("Not all parameters are specified.");
+            throw  new IllegalArgumentException("Not all parameters are specified.");
+        }
+
+        List<IslInfoData> islInfoData = getAllLinks(session);
+        List<PathNode> pathNode = new ArrayList<>();
+        pathNode.add(new PathNode(sourceSwitchId, sourcePort, 0));
+        pathNode.add(new PathNode(destinationSwitchId, destinationPort, 0));
+
+        boolean pathNodeMatch = islInfoData.stream().map(PathInfoData::getPath).anyMatch(p -> p.equals(pathNode));
+
+        if (!pathNodeMatch) {
+            logger.error("There is no ISL between {}-{} and {}-{}.",
+                    sourceSwitchId, sourcePort, destinationSwitchId, destinationPort);
+            throw new IslNotFoundException(String.format("There is no ISL between %s-%d and %s-%d.",
+                    sourceSwitchId, sourcePort, destinationSwitchId, destinationPort));
+        }
+
+        logger.debug("Processing get all flow ids for link request");
+        String q = "MATCH (:switch)-[fc:flow_segment]->(:switch)"
+                + "WHERE fc.src_switch={src_switch} "
+                + "AND fc.src_port={src_port} "
+                + "AND fc.dst_switch={dst_switch} "
+                + "AND fc.dst_port={dst_port} "
+                + "RETURN fc.flowid as flowid";
+
+        Map<String, Object> parameters = new HashMap<>();
+        String srcSwitch = Optional.ofNullable(sourceSwitchId)
```

Comment:
You have already check on null this variables, plus you can't add variable in query if it null.